src/overlays/actors/ovl_Obj_Ocarinalift/z_obj_ocarinalift.c

```diff
@@ -28,56 +43,229 @@ const ActorInit Obj_Ocarinalift_InitVars = {
     (ActorFunc)ObjOcarinalift_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80AC9D70[] = {
-    ICHAIN_U8(targetMode, 2, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_CONTINUE),
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(targetMode, 2, ICHAIN_CONTINUE),         ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE), ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void func_80AC94C0(ObjOcarinalift* this, s32 arg1) {
+    Math_Vec3s_ToVec3f(&this->dyna.actor.world.pos, &this->unk170[arg1]);
+}
+
+void ObjOcarinalift_Init(Actor* thisx, PlayState* play) {
+    Path* path;
+    ObjOcarinalift* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->dyna.actor.shape.rot.x = 0;
+    this->dyna.actor.world.rot.x = 0;
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.world.rot.z = 0;
+    DynaPolyActor_Init(&this->dyna, 1);
+    DynaPolyActor_LoadMesh(play, &this->dyna, &object_raillift_Colheader_0048D0);
+    this->unk160 = this->dyna.actor.home.rot.z * 0.1f;
+    if (1) {}
+    if (this->unk160 < 0.01f) {
+        func_80AC9680(this);
+    } else {
+        path = &play->setupPathList[OBJOCARINALIFT_GET_7F(&this->dyna.actor)];
+        this->unk168 = OBJOCARINALIFT_GET_1F(&this->dyna.actor);
+        this->unk164 = path->count - 1;
+        this->unk16C = 1;
+        this->unk170 = Lib_SegmentedToVirtual(path->points);
+        func_80AC94C0(this, this->unk168);
+        if ((OBJOCARINALIFT_GET_C(&this->dyna.actor) != 1) &&
+            (Flags_GetSwitch(play, OBJOCARINALIFT_GET_SWITCH_FLAG(&this->dyna.actor)))) {
+            func_80AC96B4(this);
+        } else {
+            func_80AC9AB8(this);
+        }
+    }
+}
+
+void ObjOcarinalift_Destroy(Actor* thisx, PlayState* play) {
+    ObjOcarinalift* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80AC9680(ObjOcarinalift* this) {
+    this->dyna.actor.flags &= ~ACTOR_FLAG_10;
+    this->actionFunc = func_80AC96A4;
+}
+
+void func_80AC96A4(ObjOcarinalift* this, PlayState* play) {
+}
 
-extern InitChainEntry D_80AC9D70[];
+void func_80AC96B4(ObjOcarinalift* this) {
+    this->actionFunc = func_80AC96D0;
+    this->dyna.actor.speedXZ = 0.0f;
+}
 
-extern UNK_TYPE D_06001E40;
-extern UNK_TYPE D_060048D0;
+void func_80AC96D0(ObjOcarinalift* this, PlayState* play) {
```

Comment:
https://decomp.me/scratch/eZROi
This seems reasonable by making an actor `thisx` temp.
Note it is also possible to match with the `temp` at all, but requires an `if(1){}` (see `ObjRaillift_Move` which is a very similar function).
Not sure what people prefer, but I like the `thisx` temp version myself.